zend_abstract_interface/methods/php5/methods.c

```diff
@@ -0,0 +1,113 @@
+#include "../methods.h"
+
+#include <Zend/zend_interfaces.h>
+#include <assert.h>
+#include <sandbox/sandbox.h>
+
+zend_class_entry *zai_class_lookup_ex(const char *cname, size_t cname_len TSRMLS_DC) {
+    if (!cname || !cname_len) return NULL;
+    zend_class_entry **ce;
+    /* Since we do not want to invoke the autoloader and we assume the caller
+     * will pass in the lowercased class name, we look up the class entry from
+     * the EG(class_table) directly versus calling zend_lookup_class_ex().
+     */
+    return (zend_hash_find(EG(class_table), cname, (cname_len + 1), (void **)&ce) == SUCCESS) ? *ce : NULL;
```

Comment:
Very nice catch! Yes, this would crash outside of a request context because the executor globals haven't been initialized for class lookups. I've updated this to use compiler globals instead and added a test for a class lookup outside of a request context. I also added to the docs for method calls stating that they must be called within a request context since that is a requirement of the Zend Engine. (3f8a405)